### ISSUE TITLE
Error in displayUnit definition in FMI 1.0 standard

### ISSUE THREAD

**Issue by modelica-trac-importer** [modelica/fmi-standard#74]
**Modified by andreas&#x2e;junghanns on 27 Mar 2014 08:08 UTC**
Display unit definition in FMI 1.0 is incorrect. Section 3.1 FMI ME specification says:
  a set of displayUnits is defined together with the conversion to the base unit according to the equation: 
  ```
  unit = gain*displayUnit + offset 
  ```
  `offset` is, e.g., needed for temperature units. The displayUnit definitions are used in the _ModelVariable_ element. Example for a definition: 
  ```
  <BaseUnit unit="rad/s"> 
  <DisplayUnitDefinition displayUnit="deg/s" gain=57.2957795130823/>
  ```

The provided equation is misleading and incorrect. It should read:
```
displayUnit = gain*unit + offset 
```
The issue had been fixed in FMI 2.0 specification and the attribute `gain` was renamed into `factor`. Section 2.2.2 of FMI 2.0 contains the equation:
```
DisplayUnit_value = factor*Unit_value + offset
```

If updating the released specification is not desirable, I suggest that an errata document (or at least a web-page) is created and provided next to the released specification for FMI 1.0. 
I also suggest to change the attribute name `factor` in FMI 2.0 back to  `gain` to make the standards compatible.

----

**Reported by iakov&#x2e;nakhimovski on 5 Mar 2012 09:38 UTC**
Display unit definition in FMI 1.0 is incorrect. Section 3.1 FMI ME specification says:
  a set of displayUnits is defined together with the conversion to the base unit according to the equation: 
  ```
  unit = gain*displayUnit + offset 
  ```
  `offset` is, e.g., needed for temperature units. The displayUnit definitions are used in the _ModelVariable_ element. Example for a definition: 
  ```
  <BaseUnit unit="rad/s"> 
  <DisplayUnitDefinition displayUnit="deg/s" gain=57.2957795130823/>
  ```

The provided equation is misleading and incorrect. It should read:
```
displayUnit = gain*unit + offset 
```
The issue had been fixed in FMI 2.0 specification and the attribute `gain` was renamed into `factor`. Section 2.2.2 of FMI 2.0 contains the equation:
```
DisplayUnit_value = factor*Unit_value + offset
```

If updating the released specification is not desirable, I suggest that an errata document (or at least a web-page) is created and provided next to the released specification for FMI 1.0. 
I also suggest to change the attribute name `factor` in FMI 2.0 back to  `gain` to make the standards compatible.

----

Migrated-From: https://trac.fmi-standard.org/ticket/74

**Comment by modelica-trac-importer** [comment]
**Modified by dietmarw on 7 Sep 2012 10:18 UTC**

**Comment by modelica-trac-importer** [comment]
**Comment by torstenblochwitz on 10 Jan 2013 07:13 UTC**
Please check, wether this ticket might be closed due to new unit handling in FMI 2.0!

**Comment by modelica-trac-importer** [comment]
**Changelog removed by torstenblochwitz on 10 Jan 2013 07:13 UTC**

**Comment by modelica-trac-importer** [comment]
**Comment by otter on 17 Feb 2013 21:09 UTC**
The only remaining issue with respect to FMI 2.0 is whether "factor" should be called "gain" as in FMI 1.0 (where it was used wrongly in an inconsistent way).

I have no strong opinion here and the tool vendors should decide.

**Comment by modelica-trac-importer** [comment]
**Comment by andreas&#46;junghanns on 27 Mar 2014 08:08 UTC**
Since nobody commented, we should leave things as they are. I close this ticket. Reopen if you think this needs a discussion.

**Comment by modelica-trac-importer** [comment]
**Comment by iakovn on 30 Apr 2014 12:44 UTC**
While there is no issue in FMI 2.0 the problem still remains in FMI 1.0 spec. I suggest to keep this open until a proper errata document is release for FMI 1.0.

**Comment by modelica-trac-importer** [comment]
**Comment by andreas&#46;junghanns on 30 Apr 2014 13:10 UTC**
The best place for such errata is currently the implementation notes (for 1.0): It exists already and contains similar issues.

Martin: Do you agree and can you find a good spot where to clarify this issue?

Iakov: Would you agree that this is for now sufficient until we might release a 1.0.1 with such clarifications and minor bug fixes?

**Comment by modelica-trac-importer** [comment]
**Comment by iakovn on 30 Apr 2014 13:17 UTC**
Replying to [comment:6 andreas.junghanns]:
> The best place for such errata is currently the implementation notes (for 1.0): It exists already and contains similar issues.
> 
> Martin: Do you agree and can you find a good spot where to clarify this issue?
> 
> Iakov: Would you agree that this is for now sufficient until we might release a 1.0.1 with such clarifications and minor bug fixes?

Sure, as long as there is a clear link from https://fmi-standard.org/downloads to this document.

**Comment by modelica-trac-importer** [comment]
**Comment by andreas&#46;junghanns on 5 May 2016 10:29 UTC**
This ticket is linked to by #370.

**Comment by modelica-trac-importer** [comment]
**Comment by aviel on 13 Jul 2017 07:20 UTC**
Fixed in the Model Exchange 1.0.1 specification, released July 10th, 2017.
